File: _Src/Freem/Freem.Web.Api.Public.Contracts/DTO/Records/Running/RunningRecordResponse.cs
using Freem.Entities.Abstractions.Relations.Collection;
using Freem.Entities.Activities.Identifiers;
using Freem.Entities.Records.Models;
using Freem.Entities.RunningRecords.Identifiers;
using Freem.Entities.Tags.Identifiers;

namespace Freem.Web.Api.Public.Contracts.DTO.Records.Running;

public sealed class RunningRecordResponse
{
    public RunningRecordIdentifier Id { get; }
    
    public DateTimeOffset StartAt { get; }
    
    public RecordName? Name { get; init; }
    public RecordDescription? Description { get; init; }
    
    public IReadOnlyRelatedEntitiesIdentifiersCollection<ActivityIdentifier> Activities { get; }
    public IReadOnlyRelatedEntitiesIdentifiersCollection<TagIdentifier> Tags { get; }
    
    public RunningRecordResponse(
        RunningRecordIdentifier id, 
        DateTimeOffset startAt,
        IReadOnlyRelatedEntitiesIdentifiersCollection<ActivityIdentifier> activities, 
        IReadOnlyRelatedEntitiesIdentifiersCollection<TagIdentifier> tags)
    {
        ArgumentNullException.ThrowIfNull(id);
        ArgumentNullException.ThrowIfNull(activities);
        ArgumentNullException.ThrowIfNull(tags);
        
        Id = id;
        StartAt = startAt;
        Activities = activities;
        Tags = tags;
    }
}

File: _Src/Freem/Freem.Web/src/clients/models/records/running/RunningRecordResponse.ts
export class RunningRecordResponse {
  public readonly id: string;
  public readonly startAt: string;
  public readonly activities: string[];

  public readonly name?: string;
  public readonly description?: string;
  public readonly tags?: string[];

  constructor(
    id: string,
    startAt: string,
    activities: string[],
    name?: string,
    description?: string,
    tags?: string[]
  ) {
    this.id = id;
    this.startAt = startAt;
    this.activities = activities;

    this.name = name;
    this.description = description;
    this.tags = tags;
  }
}